common/models/user.js

```diff
@@ -468,130 +468,143 @@ module.exports = function(User) {
+      if (err) return cb(err);
       user.verificationToken = token;
-      user.save(function(err) {
-        if (err) {
-          fn(err);
-        } else {
-          sendEmail(user);
-        }
+      user.save(options, function(err) {
+        if (err) return cb(err);
+        sendEmail(user);
       });
-    });
+    }
 
     // TODO - support more verification types
     function sendEmail(user) {
-      options.verifyHref += '&token=' + user.verificationToken;
+      verifyOptions.verifyHref += '&token=' + user.verificationToken;
 
-      options.verificationToken = user.verificationToken;
+      verifyOptions.verificationToken = user.verificationToken;
 
-      options.text = options.text || g.f('Please verify your email by opening ' +
-        'this link in a web browser:\n\t%s', options.verifyHref);
+      verifyOptions.text = verifyOptions.text || g.f('Please verify your email by opening ' +
+        'this link in a web browser:\n\t%s', verifyOptions.verifyHref);
 
-      options.text = options.text.replace(/\{href\}/g, options.verifyHref);
+      verifyOptions.text = verifyOptions.text.replace(/\{href\}/g, verifyOptions.verifyHref);
 
-      options.to = options.to || user.email;
+      verifyOptions.to = verifyOptions.to || user.email;
 
-      options.subject = options.subject || g.f('Thanks for Registering');
+      verifyOptions.subject = verifyOptions.subject || g.f('Thanks for Registering');
 
-      options.headers = options.headers || {};
+      verifyOptions.headers = verifyOptions.headers || {};
 
-      options.templateFn(options, function(err, html) {
-        if (err) {
-          fn(err);
-        } else {
-          setHtmlContentAndSend(html);
-        }
-      });
+      // argument "options" is passed depending on templateFn function requirements
+      if (verifyOptions.templateFn.length == 3) {
+        verifyOptions.templateFn(verifyOptions, options, setHtmlContentAndSend);
+      } else {
+        verifyOptions.templateFn(verifyOptions, setHtmlContentAndSend);
+      }
+
+      function setHtmlContentAndSend(err, html) {
+        if (err) return cb(err);
 
-      function setHtmlContentAndSend(html) {
-        options.html = html;
+        verifyOptions.html = html;
 
-        // Remove options.template to prevent rejection by certain
+        // Remove verifyOptions.template to prevent rejection by certain
         // nodemailer transport plugins.
-        delete options.template;
+        delete verifyOptions.template;
 
-        Email.send(options, function(err, email) {
-          if (err) {
-            fn(err);
-          } else {
-            fn(null, {email: email, token: user.verificationToken, uid: user[pkName]});
-          }
-        });
+        // argument "options" is passed depending on Email.send function requirements
+        if (Email.send.length == 3) {
+          Email.send(verifyOptions, options, handleAfterSend);
+        } else {
+          Email.send(verifyOptions, handleAfterSend);
```

Comment:
`Email.send` does not support options yet - see https://github.com/strongloop/loopback/blob/95240598b3d7d3b365eaa184f53a0ae26a6aad14/lib/connectors/mail.js#L142

It would be nice to eventually fix Email's API and add support for options too. Let's leave that out of scope of this pull request though.